.github/workflows/ubuntu-n.yaml

```
@@ -7,20 +7,286 @@ on:
     branches: [ "main" ]
   workflow_dispatch:
 
+
+env:
+  WSL_UTF8: "1"
+
 jobs:
   build:
     runs-on: [self-hosted, Windows, X64]
     steps:
       - uses: actions/checkout@v4
+
       - name: Install WSL
         run: |
-          Write-Output "Installing WSL as an optional component..."
-          # NOTE: This can be undone with:
-          # Disable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux
-          if ( $(Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux).RestartNeeded ) {
-            Write-Output "WSL was just installed, restart is required"
-            Restart-Computer
-          } else {
-            Write-Output "Switching to WSL 2"
-            Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--set-default-version 2"
+          Write-Output "Installing WSL..."
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--install --no-distribution --web-download"
+          if ( ! $? ) {
+            exit 1
```

You can consider adding error messages before exiting the program like the following:
```suggestion
          if ( ! $? ) {
            Write-Error "Failed to install WSL."
            exit 1
```

Since you exit with the same error code for probably different reasons in this step, it can be helpful (but still not necessary since you have output messages for almost each step here)

---

Done